### PR TITLE
chore(transparent-proxy): remove v1

### LINKED ISSUE #4759: remove deprecated `kumactl install transparent-proxy` flags
### Description

`--skip-resolv-conf` and `--kuma-cp-ip` are marked as deprecated and not used anymore because they leverage features that have been removed already (CP DNS).
These flags were marked as deprecated in kuma 1.8.

### PR DESCRIPTION
**Pull request by lahabana**
v1 has been deprecated for a while. It's time to remove it

Fix #4759

### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/kumactl/cmd/install/install_control_plane_test.go [hunk 1]
```diff
@@ -220,15 +220,7 @@ var _ = Describe("kumactl install control-plane", func() {
...
-		Entry("should generate Kubernetes resources with CNI plugin v1", testCase{
-			extraArgs: []string{
-				"--cni-enabled",
-				"--set",
-				"legacy.cni.enabled=true",
-			},
-			goldenFile: "install-control-plane.cni-legacy-enabled.golden.yaml",
-		}),
-		Entry("should generate Kubernetes resources with transparent v2 "+
+		Entry("should generate Kubernetes resources with ebpf"+
```
Comment (slonka) [inline app/kumactl/cmd/install/install_control_plane_test.go:223]: nitpick: the name is a bit weird now `should generate Kubernetes resources with ebpf using ebpf (experimental)`

File: app/kumactl/cmd/uninstall/uninstall_transparent_proxy_test.go [hunk 2]
```diff
@@ -37,20 +34,11 @@ var _ = Describe("kumactl install tracing", func() {
 			// when
 			err := rootCmd.Execute()
 			// then
-			Expect(err).ToNot(HaveOccurred())
+			Expect(err).To(HaveOccurred())
 			// and
-			Expect(stderr.String()).To(BeEmpty())
+			Expect(stderr.String()).To(Equal("Error: transparent proxy cleanup failed: cleanup is not supported\n"))
```
Comment (slonka) [inline app/kumactl/cmd/uninstall/uninstall_transparent_proxy_test.go:39]: we don't support uninstall in v2 TP?

**Review by slonka — COMMENTED** [review]
Not sure if we should remove it if cleanup is not supported 🤔 I'll let Bart look into this as well.

File: app/kumactl/cmd/uninstall/uninstall_transparent_proxy_test.go (hunk 2 above)
Comment (lahabana) [reply app/kumactl/cmd/uninstall/uninstall_transparent_proxy_test.go:39]: Yeah we don't IMO it's silly to run this directly on your host network...

**Review by slonka — APPROVED** [review]

File: app/kumactl/cmd/uninstall/uninstall_transparent_proxy_test.go (hunk 2 above)
Comment (slonka) [reply app/kumactl/cmd/uninstall/uninstall_transparent_proxy_test.go:39]: 👍 - maybe might make sense to put this in docs

Comment (lahabana) [reply app/kumactl/cmd/uninstall/uninstall_transparent_proxy_test.go:39]: Yes I'll add it

File: app/kumactl/cmd/install/install_transparent_proxy.go [hunk 3]
```diff
@@ -297,18 +283,7 @@ func configureTransparentProxy(cmd *cobra.Command, args *transparentProxyArgs) e
...
-
-		// best effort cleanup before we apply the rules (again?)
-		_, err := tp.Cleanup(cfg)
-		if err != nil {
-			return errors.Wrapf(err, "unable to invoke cleanup")
-		}
-	} else {
-		tp = transparentproxy.V2()
-	}
+	tp = transparentproxy.V2()
```
Comment (bartsmykla) [inline app/kumactl/cmd/install/install_transparent_proxy.go:286]: Nit: There is no V1 now, so maybe we should also change the name from `V2` to something else?

File: app/kumactl/cmd/uninstall/uninstall_transparent_proxy.go [hunk 4]
```diff
@@ -48,11 +48,7 @@ func newUninstallTransparentProxy() *cobra.Command {
 				Stdout:        cmd.OutOrStdout(),
 			}
 
-			tp := transparentproxy.V1()
-
-			if args.EbpfEnabled {
-				tp = transparentproxy.V2()
-			}
+			tp := transparentproxy.V2()
```
Comment (bartsmykla) [inline app/kumactl/cmd/uninstall/uninstall_transparent_proxy.go:51]: Nit: The same suggestion as earlier. I would prefer to change the name from `V2` as there is no `V1` anymore.

File: pkg/transparentproxy/iptables/setup.go [hunk 5]
```diff
@@ -24,3 +26,7 @@ func Setup(cfg config.Config) (string, error) {
 
 	return builder.RestoreIPTables(cfg)
 }
+
+func Cleanup(cfg config.Config) (string, error) {
+	return "", errors.New("cleanup is not supported")
```
Comment (bartsmykla) [inline pkg/transparentproxy/iptables/setup.go:31]: Nit: Maybe:
```suggestion
	return "", errors.New("cleanup is currently not supported")
```
?

**Review by bartsmykla — APPROVED** [review]
Few nits, but LGMT overall

**Pull request merged**